api/src/main/java/org/openmrs/Location.java

```diff
@@ -303,7 +303,8 @@ public Location hydrate(String locationId) {
 			return Context.getLocationService().getLocation(Integer.valueOf(locationId));
 		}
 		catch (Exception e) {
-			return new Location();
```

Comment:
What is the goal of throwing an exception instead of returning a new location object?